reinterpret_cast from 'nullptr_t' to 'const GLvoid *' (aka 'const void *') is not allowed

Getting these errors on FreeBSD 11.2 with clang-6:
```
/usr/ports/science/avogadrolibs/work/avogadrolibs-1.91.0/avogadro/rendering/meshgeometry.cpp:174:23: error: reinterpret_cast from 'nullptr_t' to 'const GLvoid *' (aka 'const void *') is not allowed
                      reinterpret_cast<const GLvoid*>(NULL));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
/usr/ports/science/avogadrolibs/work/avogadrolibs-1.91.0/avogadro/rendering/cylindergeometry.cpp:226:23: error: reinterpret_cast from 'nullptr_t' to 'const GLvoid *' (aka 'const void *') is not allowed
                      reinterpret_cast<const GLvoid*>(NULL));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

@yurivict It's interesting that FreeBSD defines `NULL` as `nullptr`. The `reinterpret_cast` does not work with `nullptr` instead of `NULL` (see error output below where I replaced `NULL` with `nullptr` on my arch linux machine).

```
/home/patrick/src/openchemistry/avogadrolibs/avogadro/rendering/meshgeometry.cpp:174:62: error: invalid cast from type ‘std::nullptr_t’ to type ‘const GLvoid*’ {aka ‘const void*’}
                       reinterpret_cast<const GLvoid*>(nullptr));
```

@cryos Maybe we could replace the whole `reinterpret_cast<const GLvoid*>(NULL)` with `nullptr`? It compiles on my arch linux machine.

--- a/avogadro/rendering/cylindergeometry.cpp
+++ b/avogadro/rendering/cylindergeometry.cpp
@@ -223,7 +223,7 @@ void CylinderGeometry::render(const Camera& camera)
   // Render the loaded spheres using the shader and bound VBO.
   glDrawRangeElements(GL_TRIANGLES, 0, static_cast<GLuint>(d->numberOfVertices),
                       static_cast<GLsizei>(d->numberOfIndices), GL_UNSIGNED_INT,
-                      reinterpret_cast<const GLvoid*>(NULL));
+                      reinterpret_cast<const GLvoid*>(0));
 
   d->vbo.release();
   d->ibo.release();

--- a/avogadro/rendering/meshgeometry.cpp
+++ b/avogadro/rendering/meshgeometry.cpp
@@ -171,7 +171,7 @@ void MeshGeometry::render(const Camera& camera)
   glDrawRangeElements(GL_TRIANGLES, 0,
                       static_cast<GLuint>(d->numberOfVertices - 1),
                       static_cast<GLsizei>(d->numberOfIndices), GL_UNSIGNED_INT,
-                      reinterpret_cast<const GLvoid*>(NULL));
+                      reinterpret_cast<const GLvoid*>(0));
 
   d->vbo.release();
   d->ibo.release();

